pkg/verifier/notation/revocationfactory.go

```diff
@@ -25,3 +33,55 @@ type RevocationFactory interface {
 	// NewValidator returns a new validator instance
 	NewValidator(revocation.Options) (revocation.Validator, error)
 }
+
+// NewFetcher returns a new fetcher instance
+func NewFetcher(httpClient *http.Client, cacheRoot string) (corecrl.Fetcher, error) {
+	crlFetcher, err := corecrl.NewHTTPFetcher(httpClient)
+	if err != nil {
+		return nil, err
+	}
+	crlFetcher.Cache, err = newFileCache(cacheRoot)
+	if err != nil {
+		return nil, err
+	}
+	return crlFetcher, nil
+}
+
+// SupportCRL checks if the certificate supports CRL
+func SupportCRL(cert *x509.Certificate) bool {
+	return cert != nil && len(cert.CRLDistributionPoints) > 0
+}
+
+// cacheCRL caches the Certificate Revocation Lists (CRLs) for the given certificates using the provided CRL fetcher.
+// It logs a warning if fetching the CRL fails but does not return an error to ensure the process is not blocked.
+func CacheCRL(ctx context.Context, certs []*x509.Certificate, crlFetcher corecrl.Fetcher) {
+	logger := logrus.WithContext(ctx)
+
+	var wg sync.WaitGroup
+	for _, cert := range certs {
+		// check if the certificate supports CRL
+		if !SupportCRL(cert) {
+			continue
+		}
+		for _, crlURL := range cert.CRLDistributionPoints {
+			wg.Add(1)
+			go func(url string) {
+				defer wg.Done()
+				if _, err := crlFetcher.Fetch(ctx, url); err != nil {
+					// Log error but do not return. Ensure unblock on CRL download failure
+					logger.Infof("failed to download CRL from %s: %v", url, err)
```

Comment:
should we add error log?

Reply:
We dont want the CRL download failure to be a blocker

Reply:
Shall we have `Errorf` to replace `Infof`